include/aspect/particle/generator/interface.h

```diff
@@ -126,6 +126,21 @@ namespace aspect
           void
           parse_parameters (ParameterHandler &prm);
 
+          void
+          parse_global_parameters (ParameterHandler &prm);
+
+          bool
+          isTimeToGenerateParticles(unsigned int timeStepNumber);
```

Comment:
See my general comment about names. Also this could move to World if you agree with my second point (I am open for discussion about that though).